app/actions/accountsActions.js

```diff
@@ -26,42 +26,72 @@ const walletHasLabel = (wallet: Object, label: string) => {
 
 export const ID = 'accounts'
 
-export const updateAccountsActions = createActions(ID, (accounts: Array<Object>) => async (): Promise<Array<Object>> => {
-  const wallet = await getWallet()
-  const newWallet = { ...wallet, accounts }
-  await setStorage(STORAGE_KEY, newWallet)
-
-  return accounts
-})
+export const updateAccountsActions = createActions(
+  ID,
+  (accounts: Array<Object>) => async (): Promise<Array<Object>> => {
+    const wallet = await getWallet()
+    const newWallet = { ...wallet, accounts }
+    await setStorage(STORAGE_KEY, newWallet)
 
-export const saveAccountActions = createActions(ID, ({ label, address, key }: Object) => async () => {
-  if (isEmpty(label)) {
-    throw new Error('A valid name is required.')
+    return accounts
   }
-
-  if (isEmpty(address)) {
-    throw new Error('A valid address is required.')
+)
+
+export const updateLabelActions = createActions(
+  ID,
+  ({ label, address }: Object) => async () => {
+    const wallet = await getWallet()
+    if (!label || !address) {
+      console.warn('updateLabelActions() invoked with invalid arguments')
+      return wallet.accounts
+    }
+    if (walletHasLabel(wallet, label)) {
+      // TODO: pop notification!
+      console.warn('A wallet with this name already exists locally')
+      return wallet.accounts
+    }
+
+    const accountToUpdate = wallet.accounts.find(
+      account => account.address === address
+    )
+    accountToUpdate.label = label
+    await setWallet(wallet)
+
+    return wallet.accounts
```

Comment:
Do we have to return accounts? I mean... this is just an update method right?

Reply:
This is what I am confused about... It is an update method but it needs to update the state of the accounts in the component. If I remove this line changes arent seen until refresh

Reply:
@evgenyboxer still getting familiar with spunky and how state management works in the app but is there a way to chain actions together... If thats the case than calling `setAccounts` after getting the new label would update the UI